example_test.go

```diff
@@ -555,10 +555,50 @@ func ExampleClient_Reboot() {
 	time.Sleep(10 * time.Second)
 
 	//re-establish the connection to the device
-	err = nukiClient.EstablishConnection(context.Background(), nukiDeviceAddr)
+	err = nukiClient.EstablishConnection(context.Background())
 	if err != nil {
 		panic(err)
 	}
 
 	//do something with the device...
 }
+
+func ExampleMonitorStateChanges() {
+	device, err := linux.NewDevice()
+	if err != nil {
+		panic(err)
+	}
+
+	nukiDeviceAddr := ble.NewAddr("54:D2:AA:BB:CC:DD")
+	nukiClient := NewClient(device, nukiDeviceAddr)
+	defer nukiClient.Close()
+
+	// make sure pairing was done with ClientIdTypeBridge
+	authId := command.AuthorizationId(111111) //load from file
+	privateKey := nacl.Key(make([]byte, 32))  //load from file
+	publicKey := nacl.Key(make([]byte, 32))   //load from file
+	nukiPublicKey := []byte{}                 //load from file
+
+	err = nukiClient.Authenticate(privateKey, publicKey, nukiPublicKey, authId)
+	if err != nil {
+		panic(err)
+	}
+
+	MonitorStateChanges(context.Background(), handleStateChange, nukiClient)
```

Comment:
for the example_test.go i would prefer to use an inline function as callback:
```suggestion
	MonitorStateChanges(context.Background(), func(ctx context.Context, c *Client) {
		fmt.Println("State change detected, loading new state...")

		defer c.Close()
		err := c.EstablishConnection(ctx)
		if err != nil {
			panic(err)
		}

		states, err := c.ReadStates(ctx)
		if err != nil {
			panic(err)
		}
		fmt.Printf("Device-State: %s\n", states.String())
	}, nukiClient)
```

Reply:
Maybe you can add that "real life" example into the README :+1: